src/ImageSharp/Common/Helpers/SimdUtils.cs

```diff
@@ -36,30 +35,24 @@ internal static Vector4 PseudoRound(this Vector4 v)
 
     /// <summary>
     /// Rounds all values in 'v' to the nearest integer following <see cref="MidpointRounding.ToEven"/> semantics.
-    /// Source:
-    /// <see>
-    ///     <cref>https://github.com/g-truc/glm/blob/master/glm/simd/common.h#L110</cref>
-    /// </see>
     /// </summary>
     /// <param name="v">The vector</param>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     internal static Vector<float> FastRound(this Vector<float> v)
     {
-        if (Avx2.IsSupported)
+        if (Avx2.IsSupported && Vector<float>.Count == Vector256<float>.Count)
```

Comment:
I don't know whether this is ever false. But I wanted the check to be a little more strict.

Reply:
It can be false even when `Avx2.IsSupported` as someone can set `DOTNET_PreferredVectorBitWidth=128`, so having an explicit check is goodness.

Notably you have `Sse41.RoundToNearestInteger` that can be used for `128-bit`, `Avx512F.RoundScale` that can be used for 512-bit, and `AdvSimd.RoundToNearest` that can be used for Arm64.

.NET 9+ defines `Vector.Round` which can be used to handle the xplat consideration without having to manually code it all (and same method exists on `Vector128/256/512`)